### ISSUE TITLE
TasteServer?

### ISSUE THREAD

**Issue by mrpotes** [tastejs/meta#9]
I was reading the history around tastejs/TasteApp#1, tastejs/TasteApp#2 and tastejs/TasteApp#4. It seems to me that not only would it be good to create a multiple-implementation front end app such as is being discussed there, but what about a multiple-implementation back-end as well? If we had a spec for an API that the front end app was going to use, we could implement that for multiple languages, such as .NET, Java+various libraries, node, Scala, etc - using Heroku/GAE/Azure for deployment.

I'm happy to invest some time in doing this, but it may influence the decision about what spec to use for TasteApp (from the suggestions put by @paulmillr and @jeremychone), which is still undecided.

What do people think?

**Comment by sindresorhus** [comment]
@mrpotes that's what https://github.com/tastejs/TasteStack was supposed to do for todo apps, but @addyosmani and I never managed to find time for it. We'll still be interested to see something happening there for sure.

**Comment by paulmillr** [comment]
@mrpotes I totally support this idea. Ost.io has an actual API specification so you can make a second backend app in .net even today.

**Comment by mrpotes** [comment]
@paulmillr We could use the ost.io API - it would probably need a bit more of a concrete definition though?

**Comment by paulmillr** [comment]
What exactly do you need?

**Comment by mrpotes** [comment]
@paulmillr sorry for the delay in replying. Probably just a brief description of what you expect each end point to actually do. For example, I think the /auth/github just does the standard OAuth flow, with client ID and a state token (that is presumably stored in a session for verification, or something?), however, that's not clear. Similarly /auth/github/callback needs defining - is this the point at which the user's list of repos is fetched? (I notice the list of repos for your account doesn't include ostio or ostio-api, for example).

@sindresorhus I saw that @kouphax has created a todomvc-server which is also TasteStack-ish - perhaps we could have TasteMVC to cover that, and TasteREST to cover multiple implementations of the ost.io API?

**Comment by kouphax** [comment]
@mrpotes My todomvc-server project(s) predates the TasteStack stuff and has been collecting dust over the past while.  Happy help revive the serve side stuff as there is a lot of interesting stuff that could be shown.

**Comment by paulmillr** [comment]
@mrpotes good idea about the details for /auth/github/callback. No problem with that.
